supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAttributes.scala

```diff
@@ -0,0 +1,124 @@
+
+object AwsBatchAttributes {
+  lazy val Logger = LoggerFactory.getLogger("AwsBatchAttributes")
+
+  private val availableConfigKeys = Set(
+    "concurrent-job-limit",
+    "root",
+    "dockerhub",
+    "dockerhub.account",
+    "dockerhub.token",
+    "filesystems",
+    "filesystems.s3.auth",
+    "filesystems.s3.caching.duplication-strategy",
+    "default-runtime-attributes",
+    "default-runtime-attributes.disks",
+    "default-runtime-attributes.memory",
+    "default-runtime-attributes.zones",
+    "default-runtime-attributes.continueOnReturnCode",
+    "default-runtime-attributes.cpu",
+    "default-runtime-attributes.noAddress",
+    "default-runtime-attributes.docker",
+    "default-runtime-attributes.queueArn",
+    "default-runtime-attributes.failOnStderr"
+  )
+
+  private val deprecatedAwsBatchKeys: Map[String, String] = Map(
+  )
+
+  private val context = "AwsBatch"
+
+  implicit val urlReader: ValueReader[URL] = StringReader.stringValueReader.map { URI.create(_).toURL }
+
+  def apply(awsConfig: AwsConfiguration, backendConfig: Config): AwsBatchAttributes = {
+    val configKeys = backendConfig.entrySet().asScala.toSet map { entry: java.util.Map.Entry[String, ConfigValue] => entry.getKey }
+    warnNotRecognized(configKeys, availableConfigKeys, context, Logger)
+
+    def warnDeprecated(keys: Set[String], deprecated: Map[String, String], context: String, logger: Logger) = {
+      val deprecatedKeys = keys.intersect(deprecated.keySet)
+      deprecatedKeys foreach { key => logger.warn(s"Found deprecated configuration key $key, replaced with ${deprecated.get(key)}") }
+    }
+
+    warnDeprecated(configKeys, deprecatedAwsBatchKeys, context, Logger)
+
+    val executionBucket: ErrorOr[String] = validate { backendConfig.as[String]("root") }
+    val filesystemAuthName: ErrorOr[String] = validate { backendConfig.as[String]("filesystems.s3.auth") }
+    val duplicationStrategy = validate { backendConfig.as[Option[String]]("filesystems.s3.caching.duplication-strategy").getOrElse("copy") match {
+      case "copy" => CopyCachedOutputs
+      case "reference" => UseOriginalCachedOutputs
+      case other => throw new IllegalArgumentException(s"Unrecognized caching duplication strategy: $other. Supported strategies are copy and reference. See reference.conf for more details.")
+    } }
+
+
+    def authAwsConfigForAwsBatchAttributes(
+                          bucket: String,
+                          authName: String,
+                          cachingStrategy: AwsBatchCacheHitDuplicationStrategy): ErrorOr[AwsBatchAttributes] = validate {
+      awsConfig.auth(authName) match {
+        case Valid(a) => AwsBatchAttributes(a, bucket, cachingStrategy)
+        case Invalid(a) => throw new IllegalArgumentException(s"AwsBatch Configuration, auth name '$a' is not valid.")
+      }
+    }
+
+    (executionBucket, filesystemAuthName, duplicationStrategy) flatMapN authAwsConfigForAwsBatchAttributes match {
+      case Valid(r) => r
```

Comment:
can use `getOrElse` and remove this line

Reply:
Yeah - that's the same problem as the comment on AwsBatchVolume.scala